### PR TITLE
IRGen: support reflection for swift 3 on COFF

### PR DESCRIPTION
**Pull request by compnerd**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

COFF has restrictions on portable section length (8 characters).  Use a FourCC
identifier for the section on COFF.  Take the opportunity to replace a
std::string parameter with a StringRef.

### CODE REVIEW

File: lib/IRGen/GenReflection.cpp [hunk 1]
```diff
@@ -705,47 +705,52 @@ class CaptureDescriptorBuilder : public ReflectionMetadataBuilder {
 };
 
 static std::string getReflectionSectionName(IRGenModule &IGM,
-                                            std::string Base) {
+                                            StringRef LonagName,
```
Comment (jckarter) [inline lib/IRGen/GenReflection.cpp]: Typo on `Long`?

Comment (compnerd) [inline lib/IRGen/GenReflection.cpp]: Yes :-(.

**Comment by compnerd** [comment]
Any objections to this?

**Comment by jckarter** [comment]
LGTM.

**Comment by jckarter** [comment]
@swift-ci Please test and merge

**Comment by gribozavr** [comment]
@compnerd Could you take a look at the test failures?

**Comment by compnerd** [comment]
Sorry about that.  Seems that I had introduced a typo in the last version that I uploaded (had extra ='s) in the new test.

**Comment by gribozavr** [comment]
@swift-ci Please test

**Pull request merged**